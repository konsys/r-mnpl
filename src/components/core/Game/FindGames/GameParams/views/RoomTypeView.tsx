import React from "react";
import { RoomType } from "../../CreateGameModal";
import { useTranslation } from "react-i18next";

export default function RoomTypeView({ type }: { type: RoomType }) {
  const { t } = useTranslation();
  return (
    <>
      {
        (type === RoomType.QUICK && t(RoomType.QUICK),
        type === RoomType.REGULAR && t(RoomType.REGULAR),
        type === RoomType.RETRO && t(RoomType.RETRO),
        type === RoomType.SHUFFLE && t(RoomType.SHUFFLE),
        type === RoomType.ROULETTE && t(RoomType.ROULETTE))
      }
    </>
  );
}
