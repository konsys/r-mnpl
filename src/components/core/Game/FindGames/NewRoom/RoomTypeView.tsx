import { Grid, Typography } from "@material-ui/core";
import {
  IRoomState,
  RoomPortalFieldType,
  RoomType,
  RoomTypeName,
} from "stores/Game/Room/NewRoomStore";

import Autorenew from "@material-ui/icons/Autorenew";
import CasinoIcon from "@material-ui/icons/Casino";
import EvStationIcon from "@material-ui/icons/EvStation";
import LocationSearchingIcon from "@material-ui/icons/LocationSearching";
import RadioIcon from "@material-ui/icons/Radio";
import React from "react";
import RestoreIcon from "@material-ui/icons/Restore";
import SendIcon from "@material-ui/icons/Send";
import ShuffleIcon from "@material-ui/icons/Shuffle";
import VideogameAssetIcon from "@material-ui/icons/VideogameAsset";
import VisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import { useTranslation } from "react-i18next";

export default function RoomTypeView({ room }: { room: IRoomState }) {
  const { t } = useTranslation();
  let name = "";
  let className = "";
  let icon = <VideogameAssetIcon />;

  switch (room.roomType) {
    case RoomType.REGULAR:
      name = RoomTypeName.REGULAR;
      className = RoomType.REGULAR;

      break;
    case RoomType.QUICK:
      name = RoomTypeName.QUICK;
      className = RoomType.QUICK;
      icon = <SendIcon />;
      break;
    case RoomType.RETRO:
      name = RoomTypeName.RETRO;
      className = RoomType.RETRO;
      icon = <RadioIcon />;
      break;
    case RoomType.ROULETTE:
      name = RoomTypeName.ROULETTE;
      className = RoomType.ROULETTE;
      icon = <LocationSearchingIcon />;
      break;
    case RoomType.SHUFFLE:
      name = RoomTypeName.SHUFFLE;
      className = RoomType.SHUFFLE;
      icon = <ShuffleIcon />;
      break;
  }

  return (
    <Grid
      container
      justify="flex-start"
      alignContent="center"
      alignItems="center"
      spacing={1}
    >
      <Grid item className={className}>
        {icon}
      </Grid>
      <Grid item className={className}>
        {t(name)}
      </Grid>
      <Grid item>
        {room.restarts && (
          <Typography color="textSecondary" variant="subtitle2">
            <Grid container alignItems="center">
              <RestoreIcon />
              {t("With restarts")}
            </Grid>
          </Typography>
        )}
      </Grid>
      <Grid item>
        {room.autostart && (
          <Typography color="textSecondary" variant="subtitle2">
            <Grid container alignItems="center">
              <Autorenew />
              {t("Game autostart")}
            </Grid>
          </Typography>
        )}
      </Grid>
      <Grid item>
        {room.privateRoom && (
          <Typography color="textSecondary" variant="subtitle2">
            <Grid container alignContent="center">
              <VisibilityOffIcon />
              {t("Private room")}
            </Grid>
          </Typography>
        )}
      </Grid>
      <Grid item>
        {room.portalType === RoomPortalFieldType.PORTAL && (
          <Typography color="textSecondary" variant="subtitle2">
            {t("With portal")}
          </Typography>
        )}
        {room.portalType === RoomPortalFieldType.ROULETTE && (
          <Typography color="textSecondary" variant="subtitle2">
            <Grid container alignItems="center">
              <CasinoIcon />
              {t("Roulette")}
            </Grid>
          </Typography>
        )}
        {room.portalType === RoomPortalFieldType.RUSSIAN_ROULETTE && (
          <Typography color="textSecondary" variant="subtitle2">
            <Grid container alignItems="center">
              <EvStationIcon />
              {t("Russian roulette")}
            </Grid>
          </Typography>
        )}
      </Grid>
    </Grid>
  );
}
