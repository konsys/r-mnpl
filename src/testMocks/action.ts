import { ICurrentAction } from "stores/Board/ActionStore";
import {
  BoardAction,
  IncomeMessageType,
  OutcomeMessageType,
} from "types/types";

export const testDoNothingAction: ICurrentAction = {
  actionId: "testActionId",
  event: {
    action: {
      _id: "testID",
      isModal: false,
      userId: 1,
      type: IncomeMessageType.DO_NOTHING,
    },
  },
};

export const testRollDicesModal: BoardAction = {
  _id: "testBoardAction",
  type: IncomeMessageType.INCOME_ROLL_DICES_MODAL,

  userId: 1,
  title: "testTitleRollDices",
  text: "testTextRollDices",
  isModal: true,
};
export const testRollDicesClicked: BoardAction = {
  _id: "testBoardAction",
  type: OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED,
  userId: 1,
  isModal: false,
};
