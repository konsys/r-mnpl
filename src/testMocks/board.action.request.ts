import { IBoardActionRequest, OutcomeMessageType } from "types/types";

export const testBoardActionRequest:IBoardActionRequest = {
    gameId:'testGameId',
    action:OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED
}