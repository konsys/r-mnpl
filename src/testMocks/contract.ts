import { IContract } from "types/types";

export const testContract: IContract = {
  fieldFromPrice: 1000,
  fieldIdsFrom: [1, 2],
  fieldIdsTo: [3],
  fieldToPrice: 1000,
  fromUserId: 1,
  moneyFrom: 1000,
  moneyTo: 1000,
  toUserId: 2,
};
