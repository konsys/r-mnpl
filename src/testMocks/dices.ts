import { IDices, IncomeMessageType } from "types/types";

export const testDices: IDices = {
    _id: "testId",
    isModal: true,
    type: IncomeMessageType.INCOME_ROLL_DICES_MODAL,
    dices: [1, 2, 3],
    dicesSum: 6,
    userId: 1,
  };
  