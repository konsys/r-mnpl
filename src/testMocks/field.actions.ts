import { testField, testMortgagedField, testOwnedField } from "./field";

import { IFieldModal } from "components/views/BoardViews/Field/FieldActions/FieldActions";

export const testFieldActions: IFieldModal = {
  ...testField,
  isActive: true,

  position: { top: 20, left: 20 },
};

export const testOwnedFieldActions: IFieldModal = {
  ...testOwnedField,
  isActive: true,

  position: { top: 20, left: 20 },
};

export const testStaredFieldActions: IFieldModal = {
  ...testOwnedField,
  isActive: true,

  position: { top: 20, left: 20 },
};

export const testMortgagedFieldActions: IFieldModal = {
  ...testMortgagedField,
  isActive: true,

  position: { top: 20, left: 20 },
};
