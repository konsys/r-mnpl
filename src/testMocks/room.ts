import {
  IRoomResponce,
  IRoom,
  PlayerRoomStatus,
  RoomPortalFieldType,
  RoomStatus,
  RoomType,
} from "stores/Game/Rooms/RoomsModel";
import { testPlayer1, testPlayer2 } from "./user";

export const testPlayingRoom: IRoom = {
  roomId: "testRoomIdtestPlayingRoom",
  creatorId: 1,
  winnerId: null,
  players: [
    { ...testPlayer1, playerRoomStatus: PlayerRoomStatus.ACITVE },
    { ...testPlayer2, playerRoomStatus: PlayerRoomStatus.ACITVE },
  ],
  createTime: new Date(),
  roomType: RoomType.REGULAR,
  playersNumber: 2,
  autostart: false,
  privateRoom: false,
  restarts: false,
  portalType: RoomPortalFieldType.NOP,
  roomStatus: RoomStatus.PLAYING,
};

export const test4Room: IRoom = {
  roomId: "testRoomIdtest4Room",
  creatorId: 1,
  winnerId: null,
  players: [
    { ...testPlayer1, playerRoomStatus: PlayerRoomStatus.ACITVE },
    { ...testPlayer2, playerRoomStatus: PlayerRoomStatus.ACITVE },
  ],
  createTime: new Date(),
  roomType: RoomType.REGULAR,
  playersNumber: 4,
  autostart: false,
  privateRoom: false,
  restarts: false,
  portalType: RoomPortalFieldType.NOP,
  roomStatus: RoomStatus.PLAYING,
};

export const testPendingRoom1: IRoom = {
  roomId: "testRoomIdtestPendingRoom1",
  creatorId: 1,
  winnerId: null,
  players: [
    { ...testPlayer1, playerRoomStatus: PlayerRoomStatus.ACITVE },
    { ...testPlayer2, playerRoomStatus: PlayerRoomStatus.ACITVE },
  ],
  createTime: new Date(),
  roomType: RoomType.REGULAR,
  playersNumber: 2,
  autostart: false,
  privateRoom: false,
  restarts: false,
  portalType: RoomPortalFieldType.NOP,
  roomStatus: RoomStatus.PENDING,
};

export const testPendingRoom2: IRoom = {
  roomId: "testRoomIdtestPendingRoom2",
  creatorId: 2,
  winnerId: null,
  players: [
    { ...testPlayer1, playerRoomStatus: PlayerRoomStatus.ACITVE },
    { ...testPlayer2, playerRoomStatus: PlayerRoomStatus.ACITVE },
  ],
  createTime: new Date(),
  roomType: RoomType.REGULAR,
  playersNumber: 2,
  autostart: false,
  privateRoom: false,
  restarts: false,
  portalType: RoomPortalFieldType.NOP,
  roomStatus: RoomStatus.PENDING,
};

export const testRoomsResponce: IRoomResponce = {
  playersInRooms: 2,
  rooms: [testPlayingRoom],
};

export const testPendingRoomsResponce: IRoomResponce = {
  playersInRooms: 2,
  rooms: [testPendingRoom1, testPendingRoom2],
};
