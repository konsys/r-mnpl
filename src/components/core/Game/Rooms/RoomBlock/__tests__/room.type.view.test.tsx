import React from "react";
import RoomTypeView from "../RoomTypeView";
import { shallow } from "enzyme";
import { testRoom } from "testMocks/room";

describe("Room type view test", () => {
  it("should render", () => {
    expect(shallow(<RoomTypeView room={testRoom} />)).toMatchSnapshot();
  });
});
