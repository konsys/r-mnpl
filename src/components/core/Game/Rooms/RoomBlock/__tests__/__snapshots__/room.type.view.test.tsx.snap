// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room type view test should render 1`] = `ShallowWrapper {}`;
